lib/livebook_web/live/hooks/sidebar_hook.ex

```diff
@@ -7,6 +7,8 @@ defmodule LivebookWeb.SidebarHook do
   def on_mount(:default, _params, _session, socket) do
     if connected?(socket) do
       Livebook.Hubs.subscribe([:crud, :connection])
+
+      Phoenix.PubSub.subscribe(Livebook.PubSub, "app_events")
```

Comment:
The SidebarHook may not be running, because we may not even have a loaded page.

My suggestion is to move the logic from `handle_event("shutdown", _params, socket)` to somewhere, such as `Livebook.Config.shutdown` (or even `Livebook`), and then invoke it directly from `LivebookApp`.

Reply:
@josevalim it's because we want to show the flash message. So maybe we should do `Livebook.Config.shutdown` and also broadcast a message to show the flash on any open pages? (I guess in either case the flash may not come through if the runtime stops quickly enough)